vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImplBase.java

```diff
@@ -67,7 +67,19 @@ protected boolean iterateNext() {
     boolean failed = failed();
     while (iter.hasNext()) {
       RouteImpl route = iter.next();
-      if (route.matches(this, mountPoint(), failed)) {
+      boolean matches;
+
+      try {
+        matches = route.matches(this, mountPoint(), failed);
+      } catch (IllegalArgumentException e) {
+        failed = true;
+        matches = false;
+        // Failure in handling failure!
+        if (log.isTraceEnabled()) log.trace("Failure in request match", e);
+        unhandledFailure(400, e, route.router());
```

Comment:
I'm concerned with this change since, to me, the code 400 conveys the idea that the server did not process the request at all. However, it seems possible to me that, in the chains of handlers, a previous handler with `useNormalisedPath` set to false could have, for example, updated a database.